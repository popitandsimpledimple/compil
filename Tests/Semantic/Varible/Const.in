const a : integer = 1;
const b : string = 'some string';
const b1 : string = 's';
const c : char = #123;
const d : double = 1.1;
const f : boolean = false;
const t : boolean = true;

begin
end.