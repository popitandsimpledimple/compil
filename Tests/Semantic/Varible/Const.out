program
   decls
      consts
         const
            a
            type
               integer
            1
      consts
         const
            b
            type
               string
            'some string'
      consts
         const
            b1
            type
               string
            's'
      consts
         const
            c
            type
               char
            #123
      consts
         const
            d
            type
               double
            1.1
      consts
         const
            f
            type
               boolean
            false
      consts
         const
            t
            type
               boolean
            true
   block
      compounds statements

 ───────────────────────────────────────────────────────────────────────────────────────────────────
 │ integer                   │ type                      │ integer                                 │
 │ char                      │ type                      │ char                                    │
 │ string                    │ type                      │ string                                  │
 │ double                    │ type                      │ double                                  │
 │ boolean                   │ type                      │ boolean                                 │
 ───────────────────────────────────────────────────────────────────────────────────────────────────

 ───────────────────────────────────────────────────────────────────────────────────────────────────
 │ without name              │ program name              │ no return type                          │
 │ a                         │ const                     │ integer                                 │
 │ b                         │ const                     │ string                                  │
 │ b1                        │ const                     │ string                                  │
 │ c                         │ const                     │ char                                    │
 │ d                         │ const                     │ double                                  │
 │ f                         │ const                     │ boolean                                 │
 │ t                         │ const                     │ boolean                                 │
 ───────────────────────────────────────────────────────────────────────────────────────────────────
